kie-server-parent/kie-server-tests/kie-server-integ-tests-common/src/main/java/org/kie/server/integrationtests/shared/KieServerDeployer.java

```diff
@@ -87,13 +83,15 @@ public static void buildAndDeployMavenProject(String basedir) {
             mvnArgs.add("-s");
             mvnArgs.add(kjarsBuildSettingsXml);
         }
-        int mvnRunResult = cli.doMain(mvnArgs.toArray(new String[mvnArgs.size()]), basedir, System.out, System.err);
 
-        Thread.currentThread().setContextClassLoader(classLoaderBak);
+        // Execute the Maven build using installed Maven binary
+        try (ProcessExecutor executor = new ProcessExecutor()) {
+            String command = mvnArgs.stream().collect(Collectors.joining(" "));
+            boolean executionSuccessful = executor.executeProcessCommand(command, Paths.get(basedir));
 
-        if (mvnRunResult != 0) {
-            throw new RuntimeException("Error while building Maven project from basedir " + basedir +
-                    ". Return code=" + mvnRunResult);
+            if (!executionSuccessful) {
```

Comment:
We need some better error messages here. This one is too generic and when it occured to me I had to debug to discover the real cause

Reply:
Ideally the real issue should be visible from the invoked process log - any error should be shown there.
Here in the code we can get just process exit value, that won't tell us much.

If it is not enough I can adjust the code to print complete process log (not just error) in case maven build fails. What do you think about it?

Reply:
If the errors are shown before it's fine, we don't need to print the log again. Thanks